notebooks/config.ipynb

```diff
@@ -964,12 +971,860 @@
+    "<!--cell 58-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "id": "988b72bc-7983-4d24-9d37-c9ba32a31ac4",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "[2024-10-16T08:00:19]     INFO - fixtures/config/base-config.nml\n",
+      "[2024-10-16T08:00:19]     INFO + fixtures/config/alt-config.nml\n",
+      "[2024-10-16T08:00:19]     INFO ---------------------------------------------------------------------\n",
+      "[2024-10-16T08:00:19]     INFO memo:            sent:  - False + True\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "False"
+      ]
+     },
+     "execution_count": 30,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "config.compare(\n",
+    "    config_1_path=Path('fixtures/config/base-config.nml'),\n",
+    "    config_2_path='fixtures/config/alt-config.nml',\n",
+    "    config_1_format='nml',\n",
+    "    config_2_format='nml'\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "655717b7-b7db-4711-9dc4-c00557e6c2df",
+   "metadata": {},
+   "source": [
+    "To see the behavior of `compare()` when key-value pairs are identical, one of the configs from above is copied in the cell below.\n",
+    "<!--cell 60-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 31,
+   "id": "ceb7f893-398b-41e4-818a-49cd036c2bfe",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "cp fixtures/config/base-config.nml tmp/config-copy.nml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "02c75cf7-9f24-44a0-97c6-abd7b44fb60f",
+   "metadata": {},
+   "source": [
+    "When these two files are compared, `True` is returned and the log reports no differences.\n",
+    "<!--cell 62-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 32,
+   "id": "8b495735-2396-436e-87db-d886ac6769fa",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "[2024-10-16T08:00:19]     INFO - fixtures/config/base-config.nml\n",
+      "[2024-10-16T08:00:19]     INFO + tmp/config-copy.nml\n",
+      "[2024-10-16T08:00:19]     INFO ---------------------------------------------------------------------\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "True"
+      ]
+     },
+     "execution_count": 32,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "config.compare(\n",
+    "    config_1_path='fixtures/config/base-config.nml',\n",
+    "    config_2_path='tmp/config-copy.nml',\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e14c0158-baa4-45d0-a86c-d9a245021229",
+   "metadata": {},
+   "source": [
+    "## Validating Configs\n",
+    "\n",
+    "The `config.validate()` function checks if a given config conforms to a specified JSON schema.\n",
```

Comment:
@christinaholtNOAA should we stick to the [phrasing](https://docs.google.com/presentation/d/1_gsEC9QB_l8qYyWTeZUY_vZt9wKcd_R7Z5xo1vVuwnw/edit#slide=id.g300d7275eb6_0_243) action instead of function? There's a couple other places this comes up. The mode being `config` the action being `validate`.

Reply:
I know from a [discussion in a previous PR](https://github.com/ufs-community/uwtools/pull/581#discussion_r1721076527) that the term "mode" wouldn't apply here since it's outside of the CLI context, but I'm not sure if the same goes for "action" or not. I'm happy to change to whichever term fits best.

Reply:
Yeah that's a great point. "action" should fall into the same context here as "mode" for the CLI. I'm happy with it as is then. I'm actually curious what the best fitting term would be now. There isn't any specific language int he docs that reference it a certain way.